kvmagent/kvmagent/plugins/tensorfusion/executor.py

```
@@ -0,0 +1,304 @@
+        if request.sm_percent_limit > 0:
+            env['TF_CUDA_SM_PERCENT_LIMIT'] = str(request.sm_percent_limit)
+
+        log_file = os.path.join(self.LOG_DIR, 'tf-worker-%s.log' % request.vm_uuid)
+        env['TF_LOG_PATH'] = log_file
+
+        logger.info('starting tensor-fusion-worker: cmd=%s, cuda_index=%d, gpu=%s' %
+                     (' '.join(cmd), cuda_index, pci_address))
+
+        try:
+            log_fd = open(log_file, 'a')
+            proc = subprocess.Popen(
```

Ruff (0.15.5)

[error] `subprocess` call: check for execution of untrusted input

(S603)